src/hotspot/share/cds/heapShared.cpp

```diff
@@ -289,6 +290,15 @@ void HeapShared::clear_root(int index) {
   }
 }
 
+bool HeapShared::is_too_large_to_archive(oop o) {
+  // TODO: To make the CDS heap mappable for all collectors, this function should
+  // reject objects that may be too large for *any* collector.
+  assert(UseG1GC, "implementation limitation");
+  size_t sz = align_up(o->size() * HeapWordSize, ObjectAlignmentInBytes);
+  size_t max = /*G1*/HeapRegion::min_region_size_in_words() * HeapWordSize;
```

Comment:
This code needs to be inside an `INCLUDE_G1GC` guard does it not?

Reply:
This code is inside `#if INCLUDE_CDS_JAVA_HEAP`, which is enabled only if `INCLUDE_G1GC` is defined.

We have other code within `INCLUDE_CDS_JAVA_HEAP` that unconditionally access G1 APIs. These will be fixed when we remove all dependencies of G1 from the CDS heap dumping code (see [JDK-8296344](https://bugs.openjdk.org/browse/JDK-8296344) )

Reply:
I see - thanks.